src/common/run_command.cpp

```diff
@@ -6,16 +6,59 @@
 
 #include <common/run_command.h>
 
+#include <logging.h>
 #include <tinyformat.h>
 #include <univalue.h>
+#include <util/string.h>
 
-#ifdef ENABLE_EXTERNAL_SIGNER
+#ifdef ENABLE_SUBPROCESS
 #include <util/subprocess.h>
-#endif // ENABLE_EXTERNAL_SIGNER
+#endif // ENABLE_SUBPROCESS
+
+#include <thread>
+
+#ifndef WIN32
+std::string ShellEscape(const std::string& arg)
+{
+    std::string escaped = arg;
+    util::ReplaceAll(escaped, "'", "'\"'\"'");
+    return "'" + escaped + "'";
+}
+#endif
+
+void RunShell(const std::string& str_command)
+{
+#ifdef ENABLE_SUBPROCESS
+    namespace sp = subprocess;
+    try {
+#ifdef WIN32
+        // Emulate _system behavior.
+        auto c = sp::Popen("cmd.exe /c " + str_command);
+#else
+        // Emulate system(3) behavior, but don't leak file descriptors.
+        auto c = sp::Popen({"/bin/sh", "-c", str_command}, sp::close_fds{true});
```

Comment:
TIL that upstream subprocess has a `shell` option that does the equivalent (https://github.com/arun11299/cpp-subprocess/blob/master/cpp-subprocess/subprocess.hpp#L1630) . But we removed it.

Unfortunately, currently it doesn't support Windows. Could revert the option and switch to using that, when it does. Or, unless they had reason to not do it in the trivial way we do, it seems easy to add.

Reply:
A related discussion takes place at https://github.com/bitcoin/bitcoin/pull/32577.

Reply:
This is a first use of the `Popen(std::initializer_list<const char*> cmd_args, Args&& ...args)` ctor. Before this, it was a candidate for cleanup.